quickjs.c

```diff
@@ -10251,7 +10251,8 @@ static JSValue js_atof2(JSContext *ctx, const char *str, const char **pp,
         p++;
     }
     if (!(flags & ATOD_INT_ONLY)) {
-        if (*p == '.' && (p > p_start || to_digit((uint8_t)p[1]) < radix)) {
+        if (*p == '.' && radix == 10 &&
+            (p > p_start || to_digit((uint8_t)p[1]) < radix)) {
```

Comment:
As far as I can tell, non-decimal floats are already denied later in the same function by this code:

```
    if (flags & ATOD_ACCEPT_SUFFIX) {
        if (*p == 'n') {
            p++;
            atod_type = ATOD_TYPE_BIG_INT;
        } else if (is_float && radix != 10) {
            goto fail;
        }
    } else if ((atod_type == ATOD_TYPE_FLOAT64) && is_float && radix != 10) {
        goto fail;
    }
```

(The `*p == 'n'` case does a bit more dance but also goes to fail on is_float.)

Anyway, it's certainly better to not do work that we just throw away later. I have removed the hex syntax parsing (and also the above is_float && radix != 10 checks, because they are now always false).

Reply:
Yes, all good.